### PR TITLE
Fixing skeleton track project export

### PR DESCRIPTION
**Pull request by Eldies**
### Motivation and context

Fixes https://github.com/cvat-ai/cvat/issues/8397

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Enhanced frame selection logic for shape conversion in the annotation process.
	- Introduced a new "skeleton track" section in the annotations JSON structure to support complex tracking scenarios.
	- Added functionality to retrieve tasks associated with specific projects via a new private method.

- **Bug Fixes**
	- Improved type clarity for the `include_empty` parameter in the `group_by_frame` method.

- **Tests**
	- Added a new test for validating the upload of annotations for tasks of type "track."

### CODE REVIEW

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 1]
```diff
@@ -345,6 +352,13 @@ def _delete_project(self, project_id, user):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         return response
 
+    @staticmethod
+    def _save_file_from_response(response, file_zip_name):
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:356]:
```suggestion
    def _save_file_from_response(response, file_name):
```

Doesn't seem to be anything zip-specific here.

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 2]
```diff
@@ -345,6 +352,13 @@ def _delete_project(self, project_id, user):
...
+        if response.status_code == status.HTTP_200_OK:
+            content = BytesIO(b"".join(response.streaming_content))
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:358]: The `BytesIO` object seems unnecessary here.

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 3]
```diff
@@ -199,6 +199,13 @@ def _get_jobs(self, task_id):
             )
         return values
 
+    def _get_tasks(self, project_id):
+        with ForceLogin(self.admin, self.client):
+            values = get_paginated_collection(lambda page:
+                self.client.get("/api/tasks?project_id={}&page={}".format(project_id, page))
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:205]:
```suggestion
                self.client.get("/api/tasks", data={"project_id": project_id, "page": page})
```

File: changelog.d/20240910_134547_dmitrii.lavrukhin_fix_skeleton_project_export.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Fixing a problem with exporting a project with skeleton tracks
```
Comment (SpecLad) [inline changelog.d/20240910_134547_dmitrii.lavrukhin_fix_skeleton_project_export.md:3]: Can you be more specific about what the problem was?

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py (hunk 1 above)
Comment (Eldies) [reply cvat/apps/dataset_manager/tests/test_rest_api_formats.py]: renamed

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py (hunk 2 above)
Comment (Eldies) [reply cvat/apps/dataset_manager/tests/test_rest_api_formats.py]: removed it

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py (hunk 3 above)
Comment (Eldies) [reply cvat/apps/dataset_manager/tests/test_rest_api_formats.py]: fixed

File: changelog.d/20240910_134547_dmitrii.lavrukhin_fix_skeleton_project_export.md (hunk 4 above)
Comment (Eldies) [reply changelog.d/20240910_134547_dmitrii.lavrukhin_fix_skeleton_project_export.md]: tried to be more specific

**Pull request merged**